build.bat

```diff
@@ -1,31 +1,52 @@
 @echo off
 rem Rime build script for msvc toolchain.
-rem 2014-12-30  Chen Gong <chen.sst@gmail.com>
+rem Maintainer: Chen Gong <chen.sst@gmail.com>
 
 setlocal
 set BACK=%CD%
 
 if exist env.bat call env.bat
 
 set OLD_PATH=%PATH%
-if defined DEV_PATH set PATH=%OLD_PATH%;%DEV_PATH%
+if defined DEVTOOLS_PATH set PATH=%OLD_PATH%;%DEVTOOLS_PATH%
 path
 echo.
 
 if not defined RIME_ROOT set RIME_ROOT=%CD%
 echo RIME_ROOT=%RIME_ROOT%
 echo.
 
+if defined BOOST_ROOT (
+  if exist "%BOOST_ROOT%\boost" goto boost_found
+)
+echo Error: Boost not found! Please set BOOST_ROOT in env.bat.
+exit /b 1
+:boost_found
 echo BOOST_ROOT=%BOOST_ROOT%
 echo.
 
-if defined CMAKE_INSTALL_PATH set PATH=%PATH%;%CMAKE_INSTALL_PATH%
+if not defined BJAM_TOOLSET (
+  set BJAM_TOOLSET=msvc-%VisualStudioVersion%

```

Comment:
Should not use `%VisualStudioVersion%` here.
The number is platform toolset, not Visual Studio version.
For example `msvc-14.1` for Visual Studio 15 2017.

https://stackoverflow.com/questions/42730478/version-numbers-for-visual-studio-2017-boost-and-cmake/42738533#42738533